### PR TITLE
Display input for KubeRay parameter and check for existence

### PR DESCRIPTION
**Pull request by ChristianZaccaria**
# Issue link

# What changes have been made

Display KubeRay parameter in release workflow and check for the existence of the chosen KubeRay release.

Previously the KubeRay parameter was not displayed
image

# Verification steps

## Checks
- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [ ] Testing is not required for this change

### CODE REVIEW

File: .github/workflows/project-codeflare-release.yml [hunk 1]
```diff
@@ -186,3 +187,15 @@ jobs:
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check if KubeRay release exists
+      run: |
+        status_code=$(curl -s -o /dev/null -w "%{http_code}" https://github.com/ray-project/kuberay/releases/tag/${{ github.event.inputs.kuberay-version }})
+        if [[ "$status_code" == "200" ]]; then
+          echo "KubeRay release with version ${{ github.event.inputs.mcad-version }} exists and available to use."
+        fi
+        echo "KubeRay release with version ${{ github.event.inputs.kuberay-version }} does not exist. Please select an existing version."
```
Comment (sutaakar) [inline .github/workflows/project-codeflare-release.yml:201]: Can you make sure that the step fails in case KubeRay version is not available?

**Comment by sutaakar** [comment]
@ChristianZaccaria Thinking about usage of `Check if KubeRay release exists`.
Should it be just informative or should it be considered as a prerequisite of the release?

**Comment by ChristianZaccaria** [comment]
> @ChristianZaccaria Thinking about usage of `Check if KubeRay release exists`. Should it be just informative or should it be considered as a prerequisite of the release?

I think in this case we will always want to specify an existing version of KubeRay to ensure the compatibility matrix is updated to the correct version we have tested. Inputting a non existing version in the matrix could cause confusion. What do you think?

**Comment by sutaakar** [comment]
In such case it may be good to fail whole pipeline in case KubeRay version is not valid.
Can you adjust other steps to require this new step to pass first before being triggered?

**Comment by ChristianZaccaria** [comment]
I added that check now. I think we don't really require the `check-kuberay-version` to be triggered for instascale or codeflare-operator as the other components will need to pass first. Should I keep the check or remove them from instascale and CFO?

**Comment by sutaakar** [comment]
You can remove it from InstaScale and operator

**Comment by ChristianZaccaria** [comment]
Updated, I think it's good to go. Thanks Karel

File: .github/workflows/project-codeflare-release.yml [hunk 2]
```diff
@@ -186,3 +189,17 @@ jobs:
...
+
+  check-kuberay-version:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check if KubeRay release exists
+      run: |
+        status_code=$(curl -s -o /dev/null -w "%{http_code}" https://github.com/ray-project/kuberay/releases/tag/${{ github.event.inputs.kuberay-version }})
+        if [[ "$status_code" == "200" ]]; then
+          echo "KubeRay release with version ${{ github.event.inputs.mcad-version }} exists and available to use."
```
Comment (sutaakar) [inline .github/workflows/project-codeflare-release.yml:201]:
```suggestion
          echo "KubeRay release with version ${{ github.event.inputs.kuberay-version }} exists and available to use."
```

**Comment by sutaakar** [comment]
One nitpick above,otherwise looks fine.

**Comment by ChristianZaccaria** [comment]
Thanks good catch!

**Pull request merged**